editoast/src/views/infra/delimited_area.rs

```diff
@@ -0,0 +1,934 @@
+    DirectedLocation,
+    InputError,
+}
+
+// Maximum distance the graph can be explored from a speed limit execution signal
+// without finding any legitimate ending to the speed limit before it is considered
+// there is not valid limit on the portion of the graph that is being explored.
+// TODO Magic number for now. Make it configurable ?
+const MAXIMUM_DISTANCE: f64 = 5000.;
+
+#[derive(Deserialize, ToSchema)]
+struct DelimitedAreaForm {
+    #[schema(inline)]
+    entries: Vec<DirectedLocation>,
+    #[schema(inline)]
+    exits: Vec<DirectedLocation>,
+}
+
+#[derive(Deserialize, Serialize, ToSchema)]
+struct DelimitedAreaResponse {
+    track_ranges: Vec<DirectionalTrackRange>,
+}
+
+#[derive(Debug, Deserialize, Serialize, ToSchema)]
+struct DirectedLocation {
+    #[schema(inline)]
+    track: Identifier,
+    position: f64,
+    direction: Direction,
+}
+
+#[derive(Debug, Error, Serialize, Deserialize, EditoastError)]
+#[editoast_error(base_id = "delimited_area")]
+enum DelimitedAreaError {
+    #[error("Some locations were invalid")]
+    #[editoast_error(status = 400)]
+    InvalidLocations {
+        invalid_locations: Vec<(DirectedLocation, InputError)>,
+    },
+}
+
+#[derive(Debug, Error, Serialize, Deserialize, ToSchema)]
+enum InputError {
+    #[error("Track '{0}' does not exist")]
+    TrackDoesNotExist(String),
+    #[error("Invalid input position '{position}' on track '{track}' of length '{track_length}'")]
+    LocationOutOfBounds {
+        track: String,
+        position: f64,
+        track_length: f64,
+    },
+}
+
+#[derive(Debug, Error, Serialize, Deserialize)]
+enum TrackRangeConstructionError {
+    #[error("Track identifiers do not match")]
+    TrackIdentifierMissmatch,
+    #[error("Track missing from infra cache")]
+    TrackNotInInfraCache,
+    #[error(
+        "The input directions or locations on track do not allow to build a valid track range"
+    )]
+    InvalidRelativeLocations,
+    #[error("The location is not on track")]
+    LocationNotOnTrack,
+}
+
+impl From<Sign> for DirectedLocation {
+    fn from(value: Sign) -> Self {
+        let Sign {
+            track,
+            position,
+            direction,
+            ..
+        } = value;
+        DirectedLocation {
+            track,
+            position,
+            direction,
+        }
+    }
+}
+
+#[utoipa::path(
+    get, path = "",
+    tag = "delimited_area",
+    params(InfraIdParam),
+    request_body = inline(DelimitedAreaResponse),
+    responses(
+        (status = 200, body = inline(DelimitedAreaResponse), description = "The track ranges between a list entries and exits." ),
+    )
+)]
+/// Return any track between one of the `entries` and one of the `exits`, i.e. any track that can
+/// be reached from an entry before reaching an exit.
+/// To prevent a missing exit to cause the graph traversal to never stop exploring, the exploration
+/// stops when a maximum distance is reached and no exit has been found.
```

Comment:
```suggestion
/// Computes all tracks between a set of `entries` locations and a set of `exits` locations
///
/// Returns any track between one of the `entries` and one of the `exits`, i.e. any track that can
/// be reached from an entry before reaching an exit.
/// To prevent a missing exit to cause the graph traversal to never stop exploring, the exploration
/// stops when a maximum distance is reached and no exit has been found.
```

Rust documentation follows this scheme:

```rust
/// One-line summary
/// 
/// Multi-line documenta-
/// -tion.
fn my_def() -> lolz
```

`utoipa` interprets this string to determine which text to assign to `description` or `summary`. If you look at the generated OpenApi you'll see the split point is a bit broken ^^